File: python/maxSum.py
def maxSum(a,len):
    curSum=0
    maxSum=0
    for i in range(0,len):
        if a[i]+maxSum<0:
            curSum=0
        else:
            curSum+=a[i]
            maxSum=max(maxSum,curSum)
    return maxSum

if __name__ == '__main__' :
    a=[6,7,2,3,-4,3]
    print maxSum(a,len(a))

File: 编程之美/MaxSum.c
#include <stdio.h>

int max(int x,int y)
{
	return (x>y)?x:y;
}

int MaxSum(int *A,int n)
{
	int nStart=A[n-1];
	int nAll=A[n-1];
	int i;
	for(i=n-2;i>=0;i--)
	{
		nStart=max(A[i],nStart+A[i]);
		nAll=max(nStart,nAll);
	}
	return nAll;
}

int main()
{
	int A[6]={0,-2,3,5,-1,2};
	printf("%d\n",MaxSum(A,6));
	return 0;
}